catalog/admin/module/catalog-admin-module-maplayers/src/main/webapp/map-layers/index.js

```diff
@@ -0,0 +1,263 @@
+  </div>
+)
+
+Error = muiThemeable()(Error)
+
+const ProviderEditor = ({ provider, error = Map(), onUpdate }) => (
+  <div style={{ padding: 16 }}>
+    <div style={{ padding: '0 16px' }}>
+      <TextField
+        onChange={(e, value) => onUpdate(value.replace(/ /g, ''), 'url')}
+        fullWidth
+        errorText={error.get('url')}
+        value={provider.get('url') || ''}
+        id='provider-url'
+        floatingLabelText='Provider URL' />
+    </div>
+    <Flexbox flex='1' style={{ padding: '0 16px' }}>
+      <Flexbox flex='5' style={{ marginRight: 20 }}>
+        <div style={{ width: '100%' }}>
+          <SelectField
+            floatingLabelText='Provider Type'
+            fullWidth
+            value={provider.get('type') || ''}
+            errorText={error.get('type')}
+            onChange={(e, i, value) => onUpdate(value, 'type')}>
+            {Object.keys(options).map((type, i) =>
+              <MenuItem
+                key={i}
+                value={type}
+                primaryText={
+                  <Flexbox flex='1' justifyContent='space-between'>
+                    <div>{options[type].label}</div>
+                    <div>
+                      {implementationSupport(Object.keys(options[type].help))}
+                    </div>
+                  </Flexbox>} />)}
+          </SelectField>
+        </div>
+      </Flexbox>
+      <Flexbox flex='1'>
+        <TextField
+          type='number'
+          step={0.01}
+          value={typeof provider.get('alpha') === 'number' ? provider.get('alpha') : ''}
+          errorText={error.get('alpha')}
+          floatingLabelText='Alpha'
+          fullWidth
+          onChange={(e, value) => {
+            if (value === '') {
+              onUpdate('', 'alpha')
+            } else {
+              const n = Number(value)
+              if (!(n < 0 || n > 1)) {
+                onUpdate(n, 'alpha')
+              }
+            }
+          }} />
+      </Flexbox>
+    </Flexbox>
+    <List>
+      <ListItem
+        primaryText='Advanced Configuration'
+        nestedItems={[
+          (options[provider.get('type')] !== undefined)
+            ? <div key='docs' style={{ textAlign: 'center', margin: '0 15px', marginBottom: 20 }}>
+              <FlatButton
+                primary
+                target='_blank'
+                label='openlayers docs'
+                disabled={options[provider.get('type')].help.ol === undefined}
+                href={options[provider.get('type')].help.ol} />
+              <FlatButton
+                primary
+                target='_blank'
+                label='cesium docs'
+                disabled={options[provider.get('type')].help.cesium === undefined}
+                href={options[provider.get('type')].help.cesium} />
+            </div> : null,
+          <div key='ace' style={{ margin: '0 15px' }}>
+            <Error errorText={error.get('buffer')}>
+              <AceEditor
```

Comment:
Changing the Provider Type threw a bunch of these warnings in the console:
```
index.js:3532 Automatically scrolling cursor into view after selection change
this will be disabled in the next version set editor.$blockScrolling = Infinity to
disable this message
```
Following the warning's recommendation and adding this attribute to this AceEditor component causes this warning to go away:
```
editorProps={{
  $blockScrolling: Infinity
}}
```